tests/test_fastavro.py

```diff
@@ -2649,3 +2649,14 @@ def test_schema_migration_should_fail_with_different_names():
 
     with pytest.raises(SchemaResolutionError):
         roundtrip(writer_schema, records, reader_schema=reader_schema)
+
+
+def test_union_of_float_and_double_keeps_precision():
```

Comment:
Could we also add a test where there is a float but no double. For example, a schema like `schema = ["float", "double"]`